modules/cxf/src/test/java/org/mule/module/cxf/CxfBasicNonBlockingTestCase.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.cxf;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.mule.module.http.api.HttpConstants.Methods;
+import static org.mule.module.http.api.client.HttpRequestOptionsBuilder.newOptions;
+import static org.mule.tck.SensingNullRequestResponseMessageProcessor.assertSameThread;
+import org.mule.DefaultMuleMessage;
+import org.mule.api.MuleMessage;
+import org.mule.api.client.MuleClient;
+import org.mule.module.http.api.client.HttpRequestOptions;
+import org.mule.module.xml.util.XMLUtils;
+import org.mule.tck.junit4.FunctionalTestCase;
+import org.mule.tck.junit4.rule.DynamicPort;
+import org.mule.transport.http.HttpConstants;
+import org.mule.util.IOUtils;
+
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.xml.transform.TransformerFactoryConfigurationError;
+
+import org.custommonkey.xmlunit.XMLUnit;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class CxfBasicNonBlockingTestCase extends FunctionalTestCase
+{
+
+    private static final HttpRequestOptions HTTP_REQUEST_OPTIONS = newOptions().method(Methods.POST.name()).build();
+
+    private String echoWsdl;
+
+    @Rule
+    public DynamicPort dynamicPort = new DynamicPort("port1");
+
+    @Override
+    protected String getConfigFile()
+    {
+        return "basic-conf-flow-httpn-nb.xml";
+    }
+
+    @Override
+    protected void doSetUp() throws Exception
+    {
+        super.doSetUp();
+        echoWsdl = IOUtils.getResourceAsString("cxf-echo-service.wsdl", getClass());
+        XMLUnit.setIgnoreWhitespace(true);
+        try
+        {
+            XMLUnit.getTransformerFactory();
+        }
+        catch (TransformerFactoryConfigurationError e)
+        {
+            XMLUnit.setTransformerFactory(XMLUtils.TRANSFORMER_FACTORY_JDK5);
+        }
+    }
+
+    @Test
+    public void testEchoService() throws Exception
```

Comment:
Not clear what you are testing/asserting here because i) there isn't a non-blocking component in the flow ii) you are asserting threads are same and not different.

Also, is it possible to paramatize an existing CXF test case?